AirCasting/ABConnector/BluetoothManager.swift

```diff
@@ -158,32 +208,53 @@ extension BluetoothManager: CBPeripheralDelegate {
                     peripheral.setNotifyValue(true, for: characteristic)
                     hasSomeCharacteristics = true
                 }
+
+                if characteristic.uuid == DOWNLOAD_FROM_SD_CARD_CHARACTERISTIC_UUID {
+                    peripheral.setNotifyValue(true, for: characteristic)
+                    hasSomeCharacteristics = true
+                }
+
+                if characteristic.uuid == DOWNLOAD_META_DATA_FROM_SD_CARD_CHARACTERISTIC_UUID {
+                    peripheral.setNotifyValue(true, for: characteristic)
+                    hasSomeCharacteristics = true
+                }
             }
         }
         hasSomeCharacteristics ? NotificationCenter.default.post(name: .discoveredCharacteristic, object: nil, userInfo: [AirCastingNotificationKeys.DiscoveredCharacteristic.peripheralUUID : peripheral.identifier]) : nil
     }
-    
+
     func peripheral(_ peripheral: CBPeripheral, didUpdateValueFor characteristic: CBCharacteristic, error: Error?) {
         guard let value = characteristic.value else {
             Log.warning("AirBeam sent measurement without value")
             return
         }
+        characteristicsMappingLock.lock()
+        charactieristicsMapping[characteristic.uuid]?.forEach { block in
```

Comment:
I'm having second thoughts here...
Thing is - we are basically blocking bluetooth for the time that those observer blocks are all executed. And we rely on this class clients to be good citizens and not take too much time. But this is not how such core app components should be built - this class should be like a tank and protect itself from all malicious scenarios. This is why I think we should have this `block.action` call scheduled to background. I will add it in another commit

Reply:
Great catch. Your solution is not working properly yet though, and the sync never completes, so I would move this commit to a separate branch and merge it in another PR. I really would like to merge this downloading functionality, so that we can continue working on other aspects of this sync.